### ISSUE TITLE
Crash during startup: kotlin.KotlinNullPointerException in onBillingSetupFinished

### ISSUE THREAD

**Issue by mzsanford** [RevenueCat/purchases-android#53]
> **Library**: `com.revenuecat.purchases:purchases:1.4.1`
> **Android Versions**: *All* (6, 7, 8, 9)
> **Devices**: *All* (Samsung, LGE, Google, etc.)

We just launched a version of our app with RevenueCat support in a beta test, with only a subset of users with access to the screens that will enable real purchases. Our code initializs RevenueCat for all users and only hides the screens in question. We immediately started to see crash reports for users not in the beta test:

```
#0. Crashed: main
       at com.revenuecat.purchases.BillingWrapper.onBillingSetupFinished(BillingWrapper.kt:187)
       at com.android.billingclient.api.BillingClientImpl$BillingServiceConnection.onServiceConnected(BillingClientImpl.java:887)
       at android.app.LoadedApk$ServiceDispatcher.doConnected(LoadedApk.java:1658)
       at android.app.LoadedApk$ServiceDispatcher$RunConnection.run(LoadedApk.java:1687)
       at android.os.Handler.handleCallback(Handler.java:789)
       at android.os.Handler.dispatchMessage(Handler.java:98)
       at android.os.Looper.loop(Looper.java:164)
       at android.app.ActivityThread.main(ActivityThread.java:6944)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)
```

From https://github.com/RevenueCat/purchases-android/blob/1.4.2/purchases/src/main/kotlin/com/revenuecat/purchases/BillingWrapper.kt#L187 this seems to originate in a log message. Is there some initialization timing problem here?

**Comment by vegaro** [comment]
Hey @mzsanford. Can you please share how you are initializing the Purchases SDK? I have seen a similar crash when initializing the Purchases SDK more than once by mistake. Doing something like this replicates the issue: 

```
        Purchases.sharedInstance = Purchases.Builder(this, PURCHASES_KEY).build()
        Purchases.sharedInstance?.listener = this
        Purchases.sharedInstance = Purchases.Builder(this, PURCHASES_KEY).build()
        Purchases.sharedInstance?.listener = this
```

Setting sharedInstance a second time before the billing client has finished connecting will nullify the billingClient (without ending the connection) and therefore when `onBillingSetupFinished` is called in that BillingWrapper, the `kotlin.KotlinNullPointerException` is thrown. 

Also, to better understand what's going on, are you using a normal implementation of BillingClient alongside with Purchases? or just Purchases? I mean the users outside of the Beta test are not using BillingClient right?

**Comment by mzsanford** [comment]
I also noticed the `close()` call when resetting in https://github.com/RevenueCat/purchases-android/pull/44 and thought that may be the case. I refactored to specifically handle double-setting the sharedInstance and we'll see if it fixes it. Given where we init I didn't expect that to happen but there may be a corner case that would allow it.

We don't have any existing use of BillingClient, we're testing RevenueCat as our first in-app purchase beta.
